assets/js/base/utils/errors.ts

```diff
@@ -0,0 +1,41 @@
+export interface ErrorObject {
+	/**
+	 * Human-readable error message to display.
+	 */
+	message: string;
+	/**
+	 * Context in which the error was triggered. That will determine how the error is displayed to the user.
+	 */
+	type: 'api' | 'general' | string;
+}
+
+type SimpleError = {
+	message: string;
+	type?: string;
+};
+
+export const formatError = async (
+	error: SimpleError | Response
+): Promise< ErrorObject > => {
+	if ( 'json' in error ) {
```

Comment:
This error formatter is covered by tests, so I feel confident to make some small runtime changes here.